### DISCUSSION TITLE
Newbie needs help with PATH settings to get OMP working on Powershell in Win11

### DISCUSSION THREAD

**Discussion by ShaguarWKL** [JanDeDobbeleer/oh-my-posh#2214] [Troubleshoot]
I've installed OMP using winget and was able to load it using run/wt or start menu context
![WindowsTerminal_2022-05-04_11-42-50](https://user-images.githubusercontent.com/104881834/166624348-6cabaafb-cbe9-4c77-8f2e-e1e131d86cdd.png)

However in Explorer context menu Open in Terminal or using Run/Powershell, I get this 
![WindowsTerminal_2022-05-04_11-43-59](https://user-images.githubusercontent.com/104881834/166624140-318195ea-40c1-437d-8801-97162f1fde33.png)

I use profile.ps1 to put in the OMP init commands and it's the only setting I have in the profile.
![Code_2022-05-04_11-45-41](https://user-images.githubusercontent.com/104881834/166624262-367f2be1-4430-4cb5-bdff-65eb72d366df.png)

Seems to be both ways of executing Powershell loads all the profiles on my system but somehow the 2nd method, which is the one I use the most often,  doesn't seem to load any of the variables or path of OMP for me. Is it because my profile.ps1 is missing a lot of things or somehow my default setup is corrupted or I am just doing this wrong?

**Comment by JanDeDobbeleer** [comment]
The winget installer (underlying one) should also add the right folders to your `PATH`. There have been previous reports it didn't for some people, and running it again seemed to fix it for most.

**Reply by ShaguarWKL** [reply]
![WindowsTerminal_2022-05-04_12-44-13](https://user-images.githubusercontent.com/104881834/166628415-5f1bb7fb-7993-4fe9-be30-db33dec4414d.png)
![WindowsTerminal_2022-05-04_12-45-12](https://user-images.githubusercontent.com/104881834/166628429-e60dc26e-a297-4658-9e35-4b6c2d610e62.png)
![WindowsTerminal_2022-05-04_12-45-48](https://user-images.githubusercontent.com/104881834/166628480-539919a8-caad-4dfb-a097-87602f97f5e8.png)

Did it twice again, and issue persists.

Should I just hardcode path into profile.ps1?

**Reply by ShaguarWKL** [reply]
adding this my profile.ps1, did the trick
$env:Path += ';C:\Users\warre\AppData\Local\Programs\oh-my-posh\bin'

really odd.

**Reply by JanDeDobbeleer** [reply]
It's weird as the installer should facilitate that. I'll try to check if something's off.

**Reply by ShaguarWKL** [reply]
Just to clarify. I'm using Win11
Opening Terminal from Start Menu Context Menu or Run/wt doesn't have this issue
But opening Terminal from Explorer Context Menu or Run/Powershell has this issue

So pretty sure installer added the PATH in but somehow something doesn't get read using the latter methods.

**Reply by JanDeDobbeleer** [reply]
Ah. That's weird. And for sure something internal to how it populates the context for the shell at that time.

**Reply by ShaguarWKL** [reply]
that's my conclusion, without digging into what flags Win11 context menu in different parts of the OS calls Terminal, I doubt we can figure this out and much easier to just hardcode the path into a custom profile.

I was wondering perhaps on your next release, you can just append that path statement into one of the standard custom profiles. I may be the minority here for OMP users being that I'm not a dev and I'm just using cmdline to automate certain file management stuff that using GUI is tedious as hell. but when I installed OMP to help me make it easier to read the output of my commands, I had none of the standard custom profiles for Powershell existing and from my bash days in another lifetime ago running datacentres, appending a path to PATH into .bashrc /.bash_profile doesn't really impact anything in the shell.

**Reply by JanDeDobbeleer** [reply]
@ShaguarWKL decided not to do this. Goes outside of the scope of oh-my-posh.  What I didn't ask if you restarted your machine? Some apps need a full reload of the device to reload the PATH.

**Reply by ShaguarWKL** [reply]
@JanDeDobbeleer yes machine was restarted

**Reply by JanDeDobbeleer** [reply]
@ShaguarWKL you should have a look at if the correct PATH entries got added (or rerun the installer for the latest version now).
